packages/fast-layouts-react/src/pane/pane.spec.tsx

```diff
@@ -0,0 +1,632 @@
+        const rendered: any = shallow(
+            <Pane resizeFrom={PaneResizeDirection.west} managedClasses={managedClasses} />
+        );
+
+        expect(rendered.prop("className")).toContain(managedClasses.pane__resizeWest);
+    });
+
+    describe("resize", () => {
+        test("should render a `resize control` when `resize` prop is true", () => {
+            const rendered: any = mount(
+                <Pane resizable={true} managedClasses={managedClasses} />
+            );
+
+            expect(rendered.find(`.${managedClasses.pane_resizeHandle}`)).toHaveLength(1);
+        });
+
+        test("should NOT render a `resize control` when `resize` prop is false", () => {
+            const rendered: any = mount(<Pane managedClasses={managedClasses} />);
+
+            expect(rendered.find(`.${managedClasses.pane_resizeHandle}`)).toHaveLength(0);
+        });
+
+        test("should NOT render a `resize control` when `resize` prop is true and `collapsed` is true", () => {
+            const rendered: any = mount(<Pane managedClasses={managedClasses} />);
+
+            expect(rendered.find(`.${managedClasses.pane_resizeHandle}`)).toHaveLength(0);
+        });
+
+        test("should have role of separator", () => {
+            const rendered: any = mount(
+                <Pane resizable={true} managedClasses={managedClasses} />
+            );
+
+            expect(
+                rendered.find(`.${managedClasses.pane_resizeHandle}`).props().role
+            ).toBe("separator");
+        });
+
+        test("should fire an onResize callback on keydown of the `resize control` when `onResize` prop is passed", (): void => {
+            const mockResize: any = jest.fn();
+            const rendered: any = mount(
+                <Pane
+                    resizable={true}
+                    onResize={mockResize}
+                    managedClasses={managedClasses}
+                />
+            );
+
+            rendered
+                .find(`.${managedClasses.pane_resizeHandle}`)
+                .simulate("keyDown", { keyCode: KeyCodes.arrowRight });
+
+            expect(mockResize).toHaveBeenCalledTimes(1);
+        });
+
+        test("should increment `width` in state when a user presses right arrow on `resize control`", () => {
+            const mockResize: any = jest.fn();
+
+            // Due to client rect call, let's reset width to 0
+            const rendered: any = mount(
+                <Pane
+                    initialWidth={0}
+                    resizable={true}
+                    onResize={mockResize}
+                    managedClasses={managedClasses}
+                />
+            );
+            const initialWidth: number = rendered.state().width;
+
+            rendered
+                .find(`.${managedClasses.pane_resizeHandle}`)
+                .simulate("keyDown", { keyCode: KeyCodes.arrowRight });
+
+            expect(rendered.state().width).toBe(initialWidth + 1);
```

Comment:
should we check the style width instead?

Reply:
Technically can't do the lifecycle here to get the style update to propagate due to internal refs being undefined.